### PR TITLE
API v2 - add support for JSON arg input

### PR DESCRIPTION
**Pull request by yhwli**
API v2 for Progress Report - JSON URL parameter
**_### Note: URL Parameter "class" is changed to "course" to maintain naming accuracy and consistency._**

Usage:

Path: /
```
URL Parameters:
    school
    course
    student_mastery
    class_mastery
```

Path: /json
  ```
Content-Type: application/json
Data:
{
    "school": "Berkeley",
    "course": "CS10",
    "student_mastery": {
        "Abstraction": 4,
        "Number Representation": 4,
        "Iteration": 3,
        "Domain and Range": 4,
        "Booleans": 1,
        "Functions": 0,
        "HOFs I": 0
    },
    "class_mastery": {
        "Abstraction": 1,
        "Number Representation": 1,
        "Iteration": 1,
        "Domain and Range": 1,
        "Booleans": 1,
        "Functions": 0,
        "HOFs I": 0
    }
```

### CODE REVIEW

File: progressReport/app.py [hunk 1]
```diff
@@ -126,17 +161,24 @@ def assign_node_levels(node, student_levels_count, class_levels_count):
...
+@app.route('/json', methods=["POST"])
+def index_json():
+    args_json = request.get_json()
+    if args_json is None:
+        return "URL JSON parameter is invalid", 400
+    school_name = args_json["school"] if "school" in args_json else "Berkeley"
+    course_name = args_json["class"] if "class" in args_json else "CS10"
+    student_mastery = args_json["student_mastery"] if "student_mastery" in args_json else {}
+    class_mastery = args_json["class_mastery"] if "class_mastery" in args_json else {}
+    return render(school_name, course_name, student_mastery, class_mastery, use_json=True)
```
Comment (Connor-Bernard) [inline progressReport/app.py:181]: The data should be submitted as a post request, but `POST` should not render HTML (even if it is possible).  This is a general rule of thumb and I mentioned it to you in person, but it doesn't seem to have been documented in the ticket directly.  [See here](https://en.wikipedia.org/wiki/Post/Redirect/Get) for more information about this design pattern.

**Review by Connor-Bernard — CHANGES_REQUESTED** [review]

File: progressReport/app.py (hunk 1 above)
Comment (william-lyh) [reply progressReport/app.py]: resolved

File: progressReport/app.py [hunk 2]
```diff
@@ -125,18 +131,62 @@ def assign_node_levels(node, student_levels_count, class_levels_count):
                            course_node_count=course_node_count,
                            course_data=course_nodes)
 
+def parse_json_mastery(school_name, course_name, student_mastery_json, class_mastery_json):
+    def assign_node_levels_json(node, student_levels_count, class_levels_count):
+        nonlocal student_mastery, class_mastery
```
Comment (Connor-Bernard) [inline progressReport/app.py:136]: Revise this to remove side effects and just return the student and class mastery

**Review by Connor-Bernard — CHANGES_REQUESTED** [review]

**Comment by Connor-Bernard** [comment]
Copied to new repo [here](https://github.com/AFA-Tooling/GradeView/pull/1)

**Pull request closed without merging**